src/server/pkg/obj/minio_client.go

```diff
@@ -23,16 +24,43 @@ func newMinioClient(endpoint, bucket, id, secret string, secure bool) (*minioCli
 	}, nil
 }
 
-func (c *minioClient) Writer(name string) (io.WriteCloser, error) {
+// Represents minio writer structure with pipe and the error channel
+type minioWriter struct {
+	errChan chan error
+	pipe    *io.PipeWriter
+}
+
+// Creates a new minio writer and a go routine to upload objects to minio server
+func newMinioWriter(client *minioClient, name string) *minioWriter {
 	reader, writer := io.Pipe()
-	go func(reader *io.PipeReader) {
-		_, err := c.PutObject(c.bucket, name, reader, "application/octet-stream")
+	w := &minioWriter{
+		errChan: make(chan error),
+		pipe:    writer,
+	}
+	go func() {
+		_, err := client.PutObject(client.bucket, name, reader, "application/octet-stream")
 		if err != nil {
 			reader.CloseWithError(err)
-			return
 		}
-	}(reader)
-	return writer, nil
+		w.errChan <- err
+	}()
+	return w
+}
+
+func (w *minioWriter) Write(p []byte) (int, error) {
+	return w.pipe.Write(p)
```

Comment:
I believe the idea is that whereas before, you'd do:

```
obj.Write(some_bytes)
Read(obj)
```

This is problematic because the fact that `Write` returns doesn't imply that the object has actually been created.

Now you'd do:

```
obj.Write(some_bytes)
obj.Close()
Read(obj)
```
And `Read` is now safe because `Close` guarantees that `PutObject` actually completes.